Makefile

```diff
@@ -107,6 +108,7 @@ else
 # Linux
 CFLAGS += -m$(MARCH) -fPIC -pthread -fno-omit-frame-pointer
 LFLAGS += -lm -Wl,-rpath,.:'$$ORIGIN':$(INSTALL_LIB_DIR) -Wl,--export-dynamic -Wl,--no-undefined
+HDLLFLAGS += -Wl,-rpath,$(INSTALL_LIB_DIR)
```

Comment:
It should be noted that if HashLink is meant to be shipped as part of a game (as in Shiro games), then the HDLLs should have an rpath that is something like `@executable_path/foo.hdll` (depends on the concrete directory structure of course). So the HDLLs will need to be patched with `install_name_tool`.

Reply:
These changes only affect Linux, right now we don't do anything rpath related on mac.

Reply:
These rpath issues have been a pain for a while. Maybe I'll open a separate issue so that we can discuss resolving them properly for all use cases.

Reply:
For now, I've added ORIGIN to the hdll's rpath so that it can load libhl from the same directory as well. This should cover the case where all the binaries are packaged in a single folder.

> then the HDLLs should have an rpath that is something like @executable_path/foo.hdll

Also note, this is about how the loader resolves libhl as a dependency of a hdll, not the other way around.